src/odemis/acq/stitching/_tiledacq.py

```diff
@@ -503,33 +508,40 @@ def estimateTime(self, remaining=None):
         if remaining is None:
             remaining = self._nx * self._ny
 
-        acq_time = 0
-        for stream in self._streams:
-            # add 1s to account for stage/objective movement time in z direction
-            acq_stream_time = acqmng.estimateTime([stream]) + 1
-            if stream.focuser is not None and len(self._zlevels) > 1:
-                # Acquisition time for each stream will be multiplied by the number of zstack levels
-                zlevels = [item for item in self._zlevels if item is not None]
-                acq_stream_time *= len(zlevels)
-            # add 2 seconds to account for switching from one tile to next tile
-            acq_time += acq_stream_time + 2
-
-        # Estimate stitching time based on number of pixels in the overlapping part
-        max_pxs = 0
-        for s in self._streams:
-            for sda in s.raw:
-                pxs = sda.shape[0] * sda.shape[1]
-                if pxs > max_pxs:
-                    max_pxs = pxs
-
-        stitch_time = (self._nx * self._ny * max_pxs * self._overlap) / self.STITCH_SPEED
+        # After the acquisition of first tile, correct the time taken for subsequent tiles based on time taken for
+        # previous tiles
+        time_per_tile = self._correct_tiled_acq_time()
+        if time_per_tile:
+            return time_per_tile * remaining
+
+        zlevels_dict = {s: self._zlevels for s in self._streams
+                        if isinstance(s, (FluoStream))}
+        acq_time = acqmng.estimateZStackAcquisitionTime(self._streams, zlevels_dict)
+        acq_time = acq_time * remaining
+
         try:
-            move_time = max(self._guessSmallestFov(self._streams)) * (remaining - 1) / self._move_speed
+            # move_speed is a default speed but not an actual stage speed due to which
+            # extra time is added based on observed time taken to move stage from one tile position to another
+            move_time = max(self._guessSmallestFov(self._streams)) * (remaining - 1) / self._move_speed + 0.3 * remaining
             # current tile is part of remaining, so no need to move there
         except ValueError:  # no current streams
             move_time = 0.5
 
-        return acq_time * remaining + move_time + stitch_time
+        logging.info(f"The computed time in seconds for tiled acquisition for {remaining} tiles for move is {move_time},"
+                     f" acquisition is {acq_time}")
+
+        return acq_time + move_time
+
+    def _correct_tiled_acq_time(self):
+        """Compute average time taken to complete tiled acquisition for one tile position"""
+        observed_time = 0
+        for k,v in self._save_time.items():
+            if not v:  # empty list => no history
+                return None
+            observed_time += numpy.mean(v)
+        if observed_time == 0:
+            return None
+        return observed_time
 
```

Comment:
Same comment as for "time_per_action". I don't see the point of storing the time into 4 different categories, and once per tile, when the only thing you do is to sum them all together.

=> Estimate the time per tile based on the starting time, and the number of tiles acquired so far.

Reply:
To get an idea how time is spent, it is saved in 4 different categories. It will be helpful incase the code for time estimation needs to be updated and actual_time_per_roi needs to be removed. I understand, saving it in 4 categories is not helpful for the calculation but it is for gaining insights later through the logs. Will that be a good way to go?

Reply:
I see. Logging in production is not very common, but I understand the need here. In that case, you should separate the logging code from the time estimation code. This way, it'll be easier to remove the logging when it's not needed. 
=> Do one single estimation of the time it took from the beginning and average based on the number of tiles acquired to estimate the time per tile (and estimate the time left).

Independently from that, record the 4 duration per tile. At the end of the function, in addition to logging the raw dict, I'd recommend also logging the average time for each type of action. This should help with interpreting the results.